stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/view/HiddenMessageListItemFilter.kt

```diff
@@ -15,6 +15,6 @@ internal object HiddenMessageListItemFilter : MessageListView.MessageListItemFil
     }
 
     override fun predicate(item: MessageListItem): Boolean {
-        return theirDeletedMessagePredicate.invoke(item).and(theirGiphyEphemeralMessagePredicate.invoke(item))
+        return theirDeletedMessagePredicate.invoke(item).and(theirGiphyEphemeralMessagePredicate.invoke(item)).not()
```

Comment:
Any chance that we can restructure the code here to make it more readable? I get that inverting the value fixes it now, but the issue is that this line is super hard to understand in the first place.

Reply:
I refactored the solution. Please, have a look